test/utils/readings.go

```diff
@@ -49,7 +50,9 @@ func WaitForReadings(t *testing.T, deviceName string, secured bool) {
 			defer resp.Body.Close()
 			require.Equal(t, 200, resp.StatusCode, "Unexpected HTTP response")
 
-			require.NoError(t, json.NewDecoder(resp.Body).Decode(&eventCount))
+			body, err := ioutil.ReadAll(resp.Body)
+			require.NoError(t, err)
+			require.NoError(t, json.Unmarshal(body, &eventCount))
```

Comment:
Sure. The original `json.NewDecoder().Decode()` method is causing an EOF error because it's trying to read from `resp.Body`, which has already been closed after the HTTP request. 

To fix this issue, this change reads `resp.Body` into a byte array `body` before decoding the JSON data. This change should prevent EOF error because the byte array is still available in memory buffer when the original response body steam has been closed.

Reply:
I don't see any code closing `resp.Body` before json decoding. The changed code look equivalent to the original.

Reply:
I believe the error shows up only when the API isn't responding with a JSON payload. It will show up both when using decode and read+unmarshal.